### PR TITLE
Align feature class naming to a new scheme.

### PR DESCRIPTION
**Pull request by jodator**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Align feature class naming to a new scheme.

BREAKING CHANGE: Renamed `HeadingEngine` to `HeadingEditing`.
BREAKING CHANGE: Extracted `HeadingUI` from `Heading`.

---

### Additional information

* Part of feature split: https://github.com/ckeditor/ckeditor5/issues/488

### CODE REVIEW

File: src/heading.js [hunk 1]
```diff
@@ -31,7 +28,7 @@ export default class Heading extends Plugin {
 	 * @inheritDoc
 	 */
 	static get requires() {
-		return [ Paragraph, HeadingEngine ];
```
Comment (scofalik) [inline src/heading.js]: I expected that `Paragraph` will be required in `Editing` but it isn't. Why is it not needed anymore? Isn't it required when applying "default option" (returning from heading to paragraph)?

File: src/headingui.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+				// Clone the option to avoid altering the original `config.heading.options`.
+				option = Object.assign( {}, option, { title } );
+			}
+
+			return option;
+		} );
+	}
+}
+
+/**
```
[the remark this answers is not in the document]
Comment (jodator) [reply src/headingui.js]: Aahhh yes. It should.

File: src/heading.js (hunk 1 above)
Comment (jodator) [reply src/heading.js]: If it's not then it looks like bug.

Comment (scofalik) [reply src/heading.js]: Ill take care of this.

File: src/headingui.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (scofalik) [reply src/headingui.js]: Ill take care of this.

[the remark this answers is not in the document]
Comment (scofalik) [reply src/headingui.js]: Now I see that it is duplicated here and there. Maybe I messed up something during merging with master.

File: src/heading.js (hunk 1 above)
Comment (scofalik) [reply src/heading.js]: `Paragraph` is required by `HeadingEditing`, so 👍. I got confused cause it has not been shown on diff.

Comment (jodator) [reply src/heading.js]: Oh, OK Then :D

**Pull request merged**